*** /usr/share/gmusicbrowser/plugins/fetch_cover.pm	2011-03-02 15:06:32.529164000 +0100
--- fetch_cover.pm	2011-03-02 15:01:46.000000000 +0100
***************
*** 28,38 ****
--- 28,40 ----
  (artist =>
   {	googlei => ['google images',"http://images.google.com/images?q=%s&imgsz=medium|large", \&parse_googlei],
  	lastfm => ['last.fm',"http://www.last.fm/music/%a/+images", \&parse_lastfm],
+     jamendo => ['Jamendo', "http://api.jamendo.com/get2/image/artist/plain/?name=%a&n=all", \&parse_jamendo],
   },
   album =>
   {	googlei => [_"google images","http://images.google.com/images?q=%s&imgsz=medium|large&imgar=ns", \&parse_googlei],
  	googleihi =>[_"google images (hi-res)","http://www.google.com/images?q=%s&imgsz=xlarge|xxlarge&imgar=ns", \&parse_googlei],
  	slothradio => ['slothradio', "http://www.slothradio.com/covers/?artist=%a&album=%l", \&parse_sloth],
+ 	jamendo => ['Jamendo', "http://api.jamendo.com/get2/image/album/plain/?artist_name=%a&n=all&imagesize=600", \&parse_jamendo],
  	#itunesgrabber => ['itunesgrabber',"http://www.thejosher.net/iTunes/index.php?artist=%a&album=%l", \&parse_itunesgrabber],
  	freecovers => ['freecovers.net', "http://www.freecovers.net/api/search/%s", \&parse_freecovers], #could add /Music+CD but then we'd lose /Soundtrack
  	#rateyourmusic=> ['rateyourmusic.com', "http://rateyourmusic.com/search?searchterm=%s&searchtype=l",\&parse_rateyourmusic], # urls results in "403 Forbidden"
***************
*** 304,309 ****
--- 306,322 ----
  	return \@list,$nexturl;
  }
  
+ sub parse_jamendo
+ {	my $result=$_[0];
+ 	my @list;
+ 
+     #print $result."\n";
+     while ($result=~m#(.+imgjam.com/[albums|artists].+)#g)
+ 	{	push @list, {url => $1};
+ 	}
+ 	return \@list;
+ }
+ 
  sub searchresults_cb
  {	my ($self,$result)=@_;
  	$self->{waiting}=undef;
